src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -473,18 +477,24 @@ def bad_move(self, order: str) -> bool:
 
     def update_allies_and_foes(self) -> None:
         power_stance = self.stance.stance[self.power_name]
+        neighbors = neighboring_opps(self.game, self.power_name, self.opponents)
 
         self.allies = [
             pow for pow in self.opponents if power_stance[pow] > self.ally_threshold
         ]
         self.foes = [
-            pow for pow in self.opponents if power_stance[pow] <= self.enemy_threshold
+            pow for pow in neighbors if power_stance[pow] <= self.enemy_threshold
         ]
+
+        # disable aggressive move filtering against allies if there are no neighboring enemies
+        if not self.foes:
+            self.is_filtering_moves = False
+        else:
+            self.is_filtering_moves = True
+
         assert self.enemy_threshold < self.ally_threshold
         self.neutral = [
-            pow
-            for pow in self.opponents
-            if self.enemy_threshold < power_stance[pow] <= self.ally_threshold
+            pow for pow in self.opponents if pow not in self.foes + self.allies
```

Comment:
Good catch! This is a smarter way to handle neutral powers.

Reply:
thanks!